src/routes/config.ts

```diff
@@ -21,13 +20,13 @@ export const vars: RequestHandler = (req, res) => {
 	const schema: JSONSchema6 = {
 		type: 'object',
 		$schema: 'http://json-schema.org/draft-06/schema#',
-		properties: {
-			...SUPERVISOR_CONFIG_VAR_PROPERTIES,
-
-			...(RESIN_HOST_CONFIG_CAPABLE_DEVICE_TYPES.includes(req.query.deviceType)
-				? HOST_CONFIG_VAR_PROPERTIES
-				: {}),
-		},
+		properties: Object.assign(
+			{},
+			SUPERVISOR_CONFIG_VAR_PROPERTIES,
+			...DEVICE_TYPE_SPECIFIC_CONFIG_VAR_PROPERTIES.filter(config =>
+				config.capableDeviceTypes.includes(req.query.deviceType),
+			).map(config => config.properties),
```

Comment:
This could be memoized if you're worried about performance

Reply:
I guess that we don't need to, since this isn't a hot endpoint, so we can save the extra memory consumption. Let me know if you have a strong opinion on this though.

Reply:
Memoizing would cause less allocations due to the filtering/mapping and so might actually be beneficial for memory - the baseline would be slightly higher but if there's multiple requests for the same device type concurrently then you'd only have one copy, and no need to worry about allocating for each request